core/src/main/java/org/openconnectors/connect/SourceConnector.java

```diff
@@ -0,0 +1,15 @@
+package org.openconnectors.connect;
+
+import org.openconnectors.util.SourceConnectorContext;
+
+import java.util.Collection;
+
+public abstract class SourceConnector<T extends SourceConnectorContext, U> extends Connector<T> {
+
+    public abstract Collection<U> poll() throws Exception;
```

Comment:
Is pull the right terminology to use?  For some sources, couldn't events be pushed to it?

Reply:
sources generate data, they are the starting point and it's poll where are you reading pull ?